thanks-bot/bot.js

```diff
@@ -11,42 +12,46 @@ logger.add(logger.transports.Console, {
 logger.level = 'debug'
 
 // Initialize Discord Bot
-const bot = new Discord.Client({
-  token: auth.token,
-  autorun: true
-})
+const client = new Discord.Client()
 
-bot.on('ready', function (evt) {
+client.on('ready', () => {
   logger.info('Connected')
   logger.info('Logged in as: ')
-  logger.info(bot.username + ' - (' + bot.id + ')')
+  logger.info(`${client.user.username} - (${client.user.id})`)
 })
 
 const ALLOWED_CHANNEL_ID = '525249065039036426'
 
-bot.on('message', function (user, userID, channelID, message, evt) {
-  let thanksMessageRegexp = /!thanks\b/gmi
+client.on('message', message => {
+  const thanksMessageRegexp = /!thanks\b/gmi
 
   // Conditions for exiting the function
-  if (channelID !== ALLOWED_CHANNEL_ID ||
-    userID === bot.id || !thanksMessageRegexp.test(message)) {
+  if (message.channel.id !== ALLOWED_CHANNEL_ID ||
+    message.author.id === client.user.id ||
+    message.content !== thanksMessageRegexp.test(message)) {
     return
   }
 
-  // Filtering userIDs
-  let mentionedUserIDs = evt.d.mentions.filter(item => item.id !== userID)
+  // mentioned Users
+  let mentionedUsers = message.mentions.users
+
+  // Filtering mentionedUserIDs
```

Comment:
We don't really need lines `38-42`. Can you solve the problem without them?

Reply:
Может быть, в тех местах кода, где я использовала переменную `mentionedUsers` писать   `message.mentions.users` ?

Reply:
Oh, sorry, you will need to use `mentionedUsernames`, but you don't need `mentionedUserIDs`, because it is used only in
```js
if (mentionedUserIDs.length === 0) {
```
and it is the same as
```js
if (mentionedUsers.length === 0) {
```

Reply:
`mentionedUserIDs` эту переменную удалить? или она потом нужна будет ?